### PR TITLE
ref(tests): Convert `TransactionThresholdModal` to RTL and TSX

### PR DESCRIPTION
**Pull request by gggritso**
This one is a little weird in a few places:

- RTL refuses to find `spinbutton` with a name, so I guess I'm just lucky there's only one?
- there's no visible loading state, so I had to manually await the mock, which I don't like
- typing into a `spinbutton` apparently saves its `value` as a string, very mysterious

### CODE REVIEW

**Review by scttcper — APPROVED** [review]

File: static/app/views/performance/transactionSummary/transactionThresholdModal.spec.tsx [hunk 1]
```diff
@@ -1,38 +1,28 @@
...
-}
-
-function clickReset(wrapper) {
-  // Click on submit.
-  const button = wrapper.find('Button[data-test-id="reset-all"] button');
-  button.simulate('click');
-  return tick();
-}
-
 function mountModal(eventView, organization, onApply) {
```
Comment (priscilawebdev) [inline static/app/views/performance/transactionSummary/transactionThresholdModal.spec.tsx:14]: can we add types here?

```suggestion
function mountModal(
  eventView: EventView,
  organization: Organization,
  onApply: React.ComponentProps<typeof TransactionThresholdModal>['onApply']
) {
```

File: static/app/views/performance/transactionSummary/transactionThresholdModal.spec.tsx [hunk 2]
```diff
@@ -1,38 +1,28 @@
...
-  return mountWithTheme(
+  render(
     <TransactionThresholdModal
-      Header={stubEl}
-      Footer={stubEl}
-      Body={stubEl}
+      Header={stubEl as ModalRenderProps['Header']}
+      Footer={stubEl as ModalRenderProps['Footer']}
+      Body={stubEl as ModalRenderProps['Body']}
+      CloseButton={stubEl as ModalRenderProps['CloseButton']}
```
Comment (priscilawebdev) [inline static/app/views/performance/transactionSummary/transactionThresholdModal.spec.tsx:20]:
```suggestion
     Body={ModalBody}
      closeModal={jest.fn()}
      CloseButton={makeCloseButton(jest.fn())}
      Header={makeClosableHeader(jest.fn())}
      Footer={ModalFooter}
```

File: static/app/views/performance/transactionSummary/transactionThresholdModal.spec.tsx [hunk 3]
```diff
@@ -1,38 +1,28 @@
-import {mountWithTheme} from 'sentry-test/enzyme';
-import {act} from 'sentry-test/reactTestingLibrary';
-import {selectByLabel} from 'sentry-test/select-new';
+import selectEvent from 'react-select-event';
 
+import {act, render, screen, userEvent, waitFor} from 'sentry-test/reactTestingLibrary';
```
Comment (priscilawebdev) [inline static/app/views/performance/transactionSummary/transactionThresholdModal.spec.tsx:3]: do we need the `act`?

File: static/app/views/performance/transactionSummary/transactionThresholdModal.spec.tsx (hunk 2 above)
Comment (priscilawebdev) [reply static/app/views/performance/transactionSummary/transactionThresholdModal.spec.tsx:20]: if you apply this suggestion, you will need to import

```
import {
  makeClosableHeader,
  makeCloseButton,
  ModalBody,
  ModalFooter,
} from 'sentry/components/globalModal/components';
```

**Comment by priscilawebdev** [comment]
> RTL refuses to find spinbutton with a name, so I guess I'm just lucky there's only one?

This is happening because we haven't defined an `aria-label` or a `htmlFor` for this field. In our code if we add an `id` to the `Field` component and the same `id` to the `Input` component, we will be able to query the role by its name 
`
<Field
          id="threshold"
          data-test-id="response-time-threshold"
          label={t('Response Time Threshold (ms)')}
          inline={false}
          help={t(
            'The satisfactory response time for the calculation method defined above. This is used to calculate Apdex and User Misery scores.'
          )}
          showHelpInTooltip
          flexibleControlStateSize
          stacked
          required
        >
          <Input
            id="threshold"
            type="number"
            name="threshold"
            required
            pattern="[0-9]*(\.[0-9]*)?"
            onChange={(event: React.ChangeEvent<HTMLInputElement>) => {
              this.handleFieldChange('threshold')(event.target.value);
            }}
            value={this.state.threshold}
            step={100}
            min={100}
          />
        </Field>
`

`
userEvent.clear(
      screen.getByRole('spinbutton', {name: 'Response Time Threshold (ms)'})
    );
`

**Comment by priscilawebdev** [comment]
> there's no visible loading state, so I had to manually await the mock, which I don't like

is this something we should add later?

**Comment by priscilawebdev** [comment]
> typing into a spinbutton apparently saves its value as a string, very mysterious

yes, that's correct. Maybe we could update the  input in another PR so that if the field is of type number, it always returns a number

File: static/app/views/performance/transactionSummary/transactionThresholdModal.spec.tsx (hunk 3 above)
Comment (gggritso) [reply static/app/views/performance/transactionSummary/transactionThresholdModal.spec.tsx:3]: Ostensibly yes, for https://github.com/getsentry/sentry/blob/99cddbe358ad7ac5eb63ec202c226597202dfa83/static/app/views/performance/transactionSummary/transactionThresholdModal.spec.tsx#L61

**Comment by gggritso** [comment]
@priscilawebdev thanks for the suggestions! I applied everything that relates to the test file, as for improvements to the component itself (adding the label, adding a loading state, formatting the input value as a number, etc.) I'll save that for another PR just to keep this PR as a pure conversion

**Review by priscilawebdev — APPROVED** [review]
left one more suggestion but it looks good to me

File: static/app/views/performance/transactionSummary/transactionThresholdModal.spec.tsx (hunk 3 above)
Comment (gggritso) [reply static/app/views/performance/transactionSummary/transactionThresholdModal.spec.tsx]: The tests pass, yeah. It's a little suspicious, `ProjectsStore.loadInitialData` is wrapped in `act` ~50% of the time in other tests 🤔

**Pull request merged**